### ISSUE TITLE
react-sdk versions >=`2.0.17` break Next.js static page generation without use of `next/dynamic`

### ISSUE THREAD

**Issue by zthart** [descope/react-sdk#505]
Issue seems to affect versions `2.0.17` -> `2.0.23`, using Next.js version `14.2.3` and the App Router. It appears importing the `AuthProvider` named export, even within components with the `'use client'` directive, breaks static page generation due to references to client-side APIs.

Broken example:
```ts
// project/app/home/layout.tsx
'use client'

import { AuthProvider } from '@descope/react-sdk'

export default function HomeLayout( {children}: {children: ReactNode} ) {
  return (
    <AuthProvider
      projectId={myProjectId}
      sessionTokenViaCookie
      baseUrl={myAppBaseUrl}
    >
      {children}
    </AuthProvider>
  );
}
```

Ran into this problem today when our dependency on `descope/react-sdk` was upgraded from `2.0.2` to `2.0.23` as part of some package upgrades (our version string in our package.json was `^2.0.2`)

After the version upgrade, we began seeing errors like this when running `next build`:

```
Error occurred prerendering page "/home". Read more: https://nextjs.org/docs/messages/prerender-error

ReferenceError: document is not defined
    /** trimmed stack trace **/
```

Using the `dynamic` utility to lazy-load the dependency seems to resolve the issue:

```ts
'use client'

import dynamic from 'next/dynamic';
// explicitly lazy-load the AuthProvider, and disable SSR
const AuthProvider = dynamic(
  () => import('@descope/react-sdk').then(mod => mod.AuthProvider),
  { ssr: false }
);

export default function HomeLayout( {children}: {children: ReactNode} ) {
  return (
    <AuthProvider
      projectId={myProjectId}
      sessionTokenViaCookie
      baseUrl={myAppBaseUrl}
    >
      {children}
    </AuthProvider>
  );
}
```

I don't fully understand why this change/regression happened, surely versions of the SDK prior to `2.0.17` _had_ to reference `document` or other properties present in the browser in order to do basic functionality? I wasn't able to find a changelog for this release that might indicate what changes were the root cause here. We've found this workaround for now, so including that here for anyone else that might run into this problem. 

I'm curious if there is a public facing changelog for the library beyond what notes are included in the github releases, which appear to be more sparse than what is published to npm. Also, are there any integration tests being run for the SDK in popular frameworks like Next.js?

**Comment by asafshen** [comment]
Hey @zthart 

Thanks for the detailed description of the issue 

I'll look into it and let you know my findings as soon as possible 

By the way, do you know about `@descope/next-sdk` package? It's a dedicated package for next which supports everything that the react sdk does

**Comment by myfrontpocket** [comment]
I'm using Remix and ran into this this morning after updating.  Tried using the NextJS SDK, but that's a no-go

**Comment by zthart** [comment]
> By the way, do you know about `@descope/next-sdk` package? It's a dedicated package for next which supports everything that the react sdk does

I wasn't aware of the Next.js-specific package. I tried it out here as a drop-in replacement and it requires the same `next/dynamic` workaround.

Looking into it it seems the version of `@descope/react-sdk` it depends on is `2.0.20`, so it seems to suffer from this same problem when using the `AuthProvider` in client components by way of the transitive dependency. I haven't had the time to investigate reorganizing some of our application to take advantage of the provided middleware to move some of this code server-side which might get around this problem in a different way.

**Comment by asafshen** [comment]
the issue was fixed on `@descope/next-js-sdk` version 0.0.10
